### PR TITLE
Display spot card when navigating to spot

### PR DESCRIPTION
**Pull request by kenzshelley**
<h1> Make a card displaying spot info appear </h1>

<h3> What does this do? </h3>

Displays a card showing a pic of the spot, the spot addr, it's rating, and the time to the spot

<h3> What does this not do </h3> 
- actually put the real spot image into the car (haven't started managing uploading images)
- send the intent to open navigation in google maps

@matthewgrossman @nickcharles

### CODE REVIEW

**Comment by nickcharles** [comment]
So, to be clear, the spot is not actually reserved and purchased by the user until they navigate to it? I.e. you would hit the "Find Spot" button, it would show you this card, and then you hit "Navigate" and it effectively serves as a confirmation and will reserve it and take you there?

File: app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java [hunk 1]
```diff
@@ -66,48 +79,127 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+            }
+        });
+
+        return view;
+    }
+
+    private void reserveSpot() {
+        requestReservation(new HttpClient.VolleyCallback<String>() {
+            @Override
+            public void onSuccess(String response) {
```
Comment (nickcharles) [inline app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java:112]: Not something for this PR, but is it possible to pull these onSuccess() functions out out this function and define them elsewhere as something like onReservationSuccess()? I'm getting a bit scared of these deeply nested callback functions.

Edit: I know this was something you addressed in PR #3 and at the time it made sense but it still seems like were ending up with nested functions. Maybe I'm missing the point of the "callback interface"

**Comment by nickcharles** [comment]
Just linking issue number #5 to this PR.

**Comment by nickcharles** [comment]
This looks really cool Kenz, great work.

**Comment by kenzshelley** [comment]
@nickcharles Yes, that will be the user's understanding of it at least. 

Technically hitting 'find spot' creates a reservation in the 'reserved' state on the backend, but that just happens so that two users can't be considering the same spot at the same time ( @matthewgrossman correct me if I'm wrong here). My understanding is that the user won't actually start paying until they've entered the 'occupied' state, which happens once they actually reach the spot.

File: app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java (hunk 1 above)
Comment (kenzshelley) [inline app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java:112]: So the basic problem is that we have a lot of requests that are dependent upon other requests, and as far as I know there isn't a pretty way to chain requests together in Java/Volley.

One approach is I think what you just suggested, which is to have one function that you call inside of the first 'onSuccess', and then inside of that function you make the next request, and so on. The problem with that is that it makes it hard to see what's going on consecutively; i.e. i want to see that I made a request to make a reservation, then I made a request to get directions to the spot reserved for me, and then I made a request to get spot information for this reservation. Having all of the requests buried inside of different functions seemed like it was going to make it harder to follow to me....but I'm not really sure. 

The nested callbacks are definitely kind of ugly too. What would be ideal is if we could do something similar similar to js promises where you can just say: 

```
requestReservation(blah).then(function(blah) {
  do stuff
  return requestThing();
}).then(function(meh) {
  do stuff;
  return requestThing2();
}).catch(function(error) {

})
```

So yeah I'm not really sure what the best approach is I guess

Comment (nickcharles) [inline app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java:112]: Let's leave it for now, but I think the case could be made for making separate functions that you would call in each onSuccess(). Sure you'd have to do some scrolling, but they'd all be in the same class right?

**Comment by nickcharles** [comment]
Unless @matthewgrossman has any doubts, let's merge this.

**Comment by matthewgrossman** [comment]
@kenzshelley yes reserved is the correct state it'll be in after hitting find spot. Occupied will be essentially when we start tracking time, which I guess isn't currently implemented

File: app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java (hunk 1 above)
Comment (kenzshelley) [inline app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java:112]: Yeah sounds good to me. I've been researching a bit more and it looks like there might be something we could do with 'blocking' volley requests that makes things a bit cleaner. That can definitely be a separate PR though
